src/bundle/Controller/Content/ContentTreeController.php

```diff
@@ -8,34 +8,58 @@
 
 namespace Ibexa\Bundle\AdminUi\Controller\Content;
 
+use Ibexa\AdminUi\Permission\LookupLimitationsTransformer;
 use Ibexa\AdminUi\REST\Value\ContentTree\LoadSubtreeRequestNode;
 use Ibexa\AdminUi\REST\Value\ContentTree\Node;
+use Ibexa\AdminUi\REST\Value\ContentTree\NodeExtendedInfo;
 use Ibexa\AdminUi\REST\Value\ContentTree\Root;
+use Ibexa\AdminUi\Specification\ContentType\ContentTypeIsUser;
 use Ibexa\AdminUi\UI\Module\ContentTree\NodeFactory;
+use Ibexa\Contracts\AdminUi\Permission\PermissionCheckerInterface;
+use Ibexa\Contracts\Core\Limitation\Target;
 use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\Repository\PermissionResolver;
+use Ibexa\Contracts\Core\Repository\Values\Content\Location;
 use Ibexa\Contracts\Core\Repository\Values\Content\Query;
+use Ibexa\Contracts\Core\Repository\Values\User\Limitation;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
 use Ibexa\Rest\Message;
 use Ibexa\Rest\Server\Controller as RestController;
 use Symfony\Component\HttpFoundation\Request;
 
+/**
+ * @phpstan-import-type TPermissionRestrictions from \Ibexa\AdminUi\REST\Value\ContentTree\NodeExtendedInfo
+ */
 class ContentTreeController extends RestController
 {
     /** @var \Ibexa\Contracts\Core\Repository\LocationService */
     private $locationService;
 
+    private PermissionCheckerInterface $permissionChecker;
+
+    private LookupLimitationsTransformer $lookupLimitationsTransformer;
+
     /** @var \Ibexa\AdminUi\UI\Module\ContentTree\NodeFactory */
     private $contentTreeNodeFactory;
```

Comment:
Given the changes to that section are extensive, maybe it's worth to introduce strict types for the remaining 2 members?